### PR TITLE
Add Zipkin Header Support

### PR DESCRIPTION
**Pull request by pamaddox**
Add flags to allow sending Zipkin headers to IRONdb nodes, which
know how to process these.

### CODE REVIEW

File: irondb/irondb.py [hunk 1]
```diff
@@ -148,12 +150,34 @@ def __init__(self):
...
+            self.zipkin_enabled = getattr(settings, 'IRONDB_ZIPKIN_ENABLED')
+        except AttributeError:
+            self.zipkin_enabled = False
+        try:
+            tl = getattr(settings, 'IRONDB_ZIPKIN_EVENT_TRACE_LEVEL')
+            if tl:
+                self.zipkin_event_trace_level = int(tl)
+                if self.zipkin_event_trace_level < 0:
+                    # Somebody tried to get cute, just disable it
+                    log.info("can't set IRONDB_ZIPKIN_EVENT_TRACE_LEVEL below zero, setting to zero\n")
```
Comment (cody271) [inline irondb/irondb.py]:
```suggestion
                    log.info("Can't set IRONDB_ZIPKIN_EVENT_TRACE_LEVEL below zero, setting to zero")
```

File: irondb/irondb.py [hunk 2]
```diff
@@ -148,12 +150,34 @@ def __init__(self):
...
+                    self.zipkin_event_trace_level = 0
+                elif self.zipkin_event_trace_level > 1:
+                    # We only support level 1 for now... may add support
+                    # for higher levels later
+                    log.info("can't set IRONDB_ZIPKIN_EVENT_TRACE_LEVEL above one, setting to one\n")
```
Comment (cody271) [inline irondb/irondb.py]:
```suggestion
                    log.info("Can't set IRONDB_ZIPKIN_EVENT_TRACE_LEVEL above one, setting to one")
```

File: irondb/irondb.py [hunk 3]
```diff
@@ -186,7 +212,14 @@ def fetch(self, query_log, start_time, end_time):
...
                         node = urls.series_multi
                         data_type = "json"
-                        d = requests.post(urls.series_multi, json = params, headers = self.headers,
+                        send_headers = copy.deepcopy(self.headers)
+                        if self.zipkin_enabled == True:
+                            traceheader = binascii.hexlify(os.urandom(8))
+                            send_headers['X-B3-TraceId'] = traceheader
+                            send_headers['X-B3-SpanId'] = traceheader
+                            if self.zipkin_event_trace_level == 1:
+                              send_headers['X-Mtev-Trace-Event'] = '1'
```
Comment (cody271) [inline irondb/irondb.py]:
```suggestion
                                send_headers['X-Mtev-Trace-Event'] = '1'
```

File: irondb/irondb.py (hunk 1 above)
Comment (pamaddox) [reply irondb/irondb.py]: Fixed

File: irondb/irondb.py (hunk 2 above)
Comment (pamaddox) [reply irondb/irondb.py]: Fixed

File: irondb/irondb.py (hunk 3 above)
Comment (pamaddox) [reply irondb/irondb.py]: Fixed

**Pull request merged**